### DISCUSSION TITLE
Question: How to decode a call?

### DISCUSSION THREAD

**Discussion by BulatSaif** [polkascan/py-substrate-interface#361] [Q&A] [answered]
Polkadot js have a tab to decode call, it accepts string and returns call object, is something similar possible with py-substrate-interface?
![Screenshot from 2023-10-30 20-05-33](https://github.com/polkascan/py-substrate-interface/assets/24387396/4c6bcddd-90bb-4cb3-af88-5203a6bbd204)

**Comment by arjanz** [comment, marked as answer]
Of course, you can manually create an `Extrinsic` SCALE-object (in PolkadotJS apps it's titled "hex-encoded call" but in reality this is an "hex-encoded extrinsic", which of course contains the call. I personally think this is a bit misleading) and then decode the hex-encoded string. Something like:

```python
extrinsic_obj = substrate.create_scale_object("Extrinsic")
extrinsic_obj.decode(ScaleBytes("0x280403000b8052d3818b01"))

print(extrinsic_obj.value)
```

And If you only want to decode the actual `Call` with corresponding hex-encoded string:

```python
call_obj = substrate.create_scale_object("Call")
call_obj.decode(ScaleBytes("0x03000b8052d3818b01"))

print(call_obj.value)
```

**Comment by BulatSaif** [comment]
@arjanz, it works, thank you for explaining!
